### PR TITLE
colobot: init at 0.1.12-alpha

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

Initial integration without the game musics to reduce the size of the
game. It works without them

Data are in another derivation to let them live their live independently
as they are quite big.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/colobot/data.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      touch $po
+    done
+    # skip music
+    rm localSrc/music/CMakeLists.txt
+    cd localSrc
+  '';
+
+  meta = with lib; {
+    homepage = "https://colobot.info/";
+    description = "Colobot: Gold Edition is a real-time strategy game, where you can program your bots";
```
Comment (RaghavSood) [inline pkgs/games/colobot/data.nix:35]: Description for the data package should be specific, helps avoid people installing the wrong one

File: pkgs/games/colobot/data.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ stdenv, lib, fetchFromGitHub, cmake
```
Comment (SuperSandro2000) [inline pkgs/games/colobot/data.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, cmake
```

We don't need lib and stdenv.lib.

File: pkgs/games/colobot/data.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  buildInputs = [ gettext ];
+
+  enableParallelBuilding = false;
+  # Build procedure requires the data folder
+  patchPhase = ''
+    cp -r $src localSrc
+    find -type d -exec chmod +w {} \;
+    for po in localSrc/help/{bots,cbot,object,generic,programs}/po/* localSrc/levels/*{/*/*,}/po/*; do
+      rm $po
+      touch $po
```
Comment (SuperSandro2000) [inline pkgs/games/colobot/data.nix:26]:
```suggestion
      echo >$po
```

File: pkgs/games/colobot/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, lib, fetchFromGitHub, cmake, boost, SDL2, SDL2_image, SDL2_ttf, libpng
+, glew, gettext, libsndfile, libvorbis, libogg, physfs, openal
+, xmlstarlet, doxygen, python3, colobot-data }:
+
+stdenv.mkDerivation rec {
+  pname = "colobot";
+  version = "0.1.12-alpha";
```
Comment (SuperSandro2000) [inline pkgs/games/colobot/default.nix:7]: We should probably add a comment to also update the data derivation and vice versa.

File: pkgs/games/colobot/data.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    # skip music
+    rm localSrc/music/CMakeLists.txt
+    cd localSrc
+  '';
+
+  meta = with lib; {
+    homepage = "https://colobot.info/";
+    description = "Colobot: Gold Edition is a real-time strategy game, where you can program your bots";
+    licenses = licenses.gpl3;
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/games/colobot/data.nix:37]: Why is this different from the game?
```suggestion
    platforms = platforms.unix;
```

Also maintainers are missing in both.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103077` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>colobot</li>
    <li>colobot-data</li>
  </ul>
</details>

File: pkgs/games/colobot/data.nix (hunk 3 above)
Comment (freezeboy) [reply pkgs/games/colobot/data.nix:27]: This one is breaking the build, it is not equivalent

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -186,6 +186,10 @@ in
 
   corgi = callPackage ../development/tools/corgi { };
 
+  colobot = callPackage ../games/colobot {};
+
+  colobot-data = callPackage ../games/colobot/data.nix {};
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:191]: does this need to be exposed as a top-level attr?

Comment (jonringer) [reply pkgs/top-level/all-packages.nix:191]: could just do something like:
```
let
  colobot-data = callPackage ./data.nix { };
in
...
```
from default.nix

Comment (freezeboy) [reply pkgs/top-level/all-packages.nix:191]: You mean the data?

Don't know if there is another pattern

File: pkgs/games/colobot/data.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/games/colobot/data.nix:27]: If you want to clear the contents of the file it should be.

File: pkgs/top-level/all-packages.nix (hunk 6 above)
Comment (jonringer) [reply pkgs/top-level/all-packages.nix]: Yea, I meant the data

Comment (freezeboy) [reply pkgs/top-level/all-packages.nix]: Ok pushed an update

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103077` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>colobot</li>
  </ul>
</details>

**Comment by freezeboy** [comment]
Why is it broken ?

File: pkgs/games/colobot/data.nix (hunk 3 above)
Comment (doronbehar) [reply pkgs/games/colobot/data.nix:27]: IIRC these are not identical.

**Comment by freezeboy** [comment]
I don't understand why ofborg is not happy, looks like an unrelated problem

**Comment by doronbehar** [comment]
The ofborg error is not a problem with ofborg. There's a policy that within nixpkgs, all packages should be able to evaluate without using declarations from `aliases.nix` - as that's usually a place where attributes are declared only for backwards compatibility. In your case, you are using `xlibs` which is an alias:

https://github.com/NixOS/nixpkgs/pull/84939

So, use `xorg`, not `xlibs`.

**Comment by freezeboy** [comment]
Oh ok, didn't know that, is there a way to check it locally as nix-build succeeds anyway ?

**Comment by freezeboy** [comment]
apparently was caused by another package

File: pkgs/games/colobot/default.nix (hunk 4 above)
Comment (doronbehar) [reply pkgs/games/colobot/default.nix:10]: @freezeboy Ideally the comment saying updates to this derivation should go along with updates to `./data.nix`, should be here, right above the version, and it shouldn't have typos :)

File: pkgs/games/colobot/default.nix [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+let
+  colobot-data = callPackage ./data.nix {};
+in
+stdenv.mkDerivation rec {
+  pname = "colobot";
+  version = "0.1.12-alpha";
+
+  # Mayb require an update tp package colobot-data as well
+  # in file data.nix next to this one
```
Comment (doronbehar) [inline pkgs/games/colobot/default.nix:13]: So I'd remove these and rewrite it upwards:

```suggestion
```

**Comment by doronbehar** [comment]
> Oh ok, didn't know that, is there a way to check it locally as nix-build succeeds anyway ?

This one is easy:

```
nix-env -qa --json --file . --arg config '{ allowAliases = false; }'
```

File: pkgs/games/colobot/default.nix (hunk 7 above)
Comment (freezeboy) [reply pkgs/games/colobot/default.nix:13]: What do you mean "upwards"?

Comment (doronbehar) [reply pkgs/games/colobot/default.nix]: > What do you mean "upwards"?

Right above `version = "...";`.

**Comment by freezeboy** [comment]
Just rebased the commit, I think it should be ok

**Review by doronbehar — APPROVED** [review]

**Pull request merged**